unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/GLTF/Tests/AnyAssetPromiseShould.cs

```diff
@@ -191,5 +190,38 @@ public IEnumerator NotTryToLoadAfterForget()
 
             keeper.Cleanup();
         }
+
+        [UnityTest]
+        public IEnumerator WaitForGLTFtoBeLoadedBeforeDestroyingIt()
+        {
+            var parentGO = new GameObject("GLTFParent");
+            var keeper = new AssetPromiseKeeper_GLTF();
+
+            string url = TestAssetsUtils.GetPath() + "/GLB/Trunk/Trunk.glb";
+            bool finishedLoading = false;
+
+            AssetPromise_GLTF promise = new AssetPromise_GLTF(scene.contentProvider, url);
+            promise.settings.parent = parentGO.transform;
+            keeper.Keep(promise);
+
+            GLTFComponent gltfComponent = promise.asset.container.GetComponentInChildren<GLTFComponent>();
+            gltfComponent.OnSuccess += () => finishedLoading = true;
+            GameObject gltfContainer = promise.asset.container;
+
+            Assert.IsNotNull(gltfComponent);
+            Assert.AreEqual(parentGO.transform, gltfContainer.transform.parent);
+
+            keeper.Forget(promise);
+            Assert.IsNotNull(gltfComponent);
+            Assert.AreNotEqual(parentGO.transform, gltfContainer.transform.parent);
+
+            yield return promise;
+
+            yield return new WaitUntil(() => finishedLoading, 2);
```

Comment:
Please make sure that this test is not flaky. If the CI takes more than 2 seconds to load the GLTF it may fail when it shouldn't.

Reply:
Done. I subscribe to fail event instead of using a timeout